ext/afform/core/Civi/Api4/Action/Afform/AbstractProcessor.php

```diff
@@ -723,8 +723,8 @@ public function processFormData(array $entityValues) {
    * @return int|string|null
    */
   protected static function getNestedKey(array $values) {
-    $firstValue = \CRM_Utils_Array::first(array_filter($values));
-    return is_array($firstValue) && $firstValue ? array_keys($firstValue)[0] : NULL;
+    $firstNonEmptyValue = \CRM_Utils_Array::first(array_filter($values));
+    return is_array($firstNonEmptyValue) ? array_keys($firstNonEmptyValue)[0] : NULL;
```

Comment:
I can't see a neater way than

```suggestion
  $nonEmptyValues = array_filter($values);

  $first = $nonEmptyValues[0] ?? NULL;

  if ($first && is_array($first)) {
    return array_keys($first)[0];
  }

  return NULL;
```